### PR TITLE
Way to manually put out controlled fire in a fireplace.

### LINKED ISSUE #32492: Extinguish controlled fires
**Is your feature request related to a problem? Please describe.**  
With new firewood source location, automatic actions can cause extended use of fire when not needed, using fuel, or requiring a fire extinguisher or dumping precious water to put out.

**Describe the solution you'd like**  
Controlled fires in a brazier, oven or other crafted structure should be extinguishable, without water, or a fire extinguisher by smothering with wood ash or other simple methods. Perhaps gate it behind survival skill 1, or require 100 wood ash to be present or something, definitely limit it to controlled fires. Takes around 5 minutes.

**Additional context**  
Already burnt wood ash is frequently used in fireplaces to smother later fires, removing oxygen. Small controlled fires can easily be put out by spreading out burning embers cooling the fire quickly, it's certainly an acquired skill but not a very difficult one.

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Features "Way to manually put out controlled fire in a fireplace"```

#### Purpose of change
Resolves #32492
Basically what the title says.

#### Describe the solution
It's achievable by examining a fireplace (brazier, etc.).
You need to wield something with `bashing` to do it. If you don't wield such thing the option will be greyed out.

#### Describe alternatives you've considered
Restricting tools so no wacky items would be used, but it's not necessary for such a petty purpose.

#### Additional context
With bashing tool:
image

Without bashing tool:
image

image

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -2370,6 +2364,10 @@ void iexamine::fireplace( player &p, const tripoint &examp )
         if( has_bionic_firestarter ) {
             selection_menu.addentry( 2, true, 'b', _( "Use a CBM to start a fire" ) );
         }
+    } else if( p.weapon.damage_melee( DT_BASH ) ) {
+        selection_menu.addentry( 4, true, 's', _( "Put out fire" ) );
+    } else {
+        selection_menu.addentry( 4, false, 's', _( "Put out fire - wield bashing item" ) );
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Maybe something like
```suggestion
        selection_menu.addentry( 4, false, 's', _( "Put out fire (bashing item required)" ) );
```

Comment (glenmack) [reply src/iexamine.cpp]: Excellent stuff!

**Comment by glenmack** [comment]
Excellent stuff!

File: src/iexamine.cpp (hunk 1 above)
Comment (nexusmrsep) [reply src/iexamine.cpp]: I deliberately want the player to choose an item by wielding it, rather then using random bashing item from inventory. Maybe it's less convenient but it seemed more natural to me.

**Comment by FulcrumA** [comment]
Doesn't unloading water container onto the tile work?

**Comment by glenmack** [comment]
> Doesn't unloading water container onto the tile work?

It does work, but it's not the best use of water and you'll probably need to mop the area before lighting a fire in that spot again, or it'll go out very quickly. There's also a lot more key presses involved and the water is very precious early game.

**Pull request merged**